hera_cal/frf.py

```diff
@@ -16,11 +16,85 @@
 from pyuvdata import UVData, UVFlag
 import argparse
 from . import io
+from . import vis_clean
 import warnings
+import astropy.constants as const
 
 
-def timeavg_waterfall(data, Navg, flags=None, nsamples=None, wgt_by_nsample=True, 
-                      wgt_by_favg_nsample=False, rephase=False, lsts=None, freqs=None, 
+SPEED_OF_LIGHT = const.c.si.value
+SDAY_KSEC = 86163.93 / 1000.
+
+
+def sky_frates(uvd, keys=None, frate_standoff=0.0, frate_width_multiplier=1.0, min_frate_half_width=0.025):
+    """Automatically compute sky fringe-rate ranges based on baselines and telescope location.
+
+    Parameters
+    ----------
+    uvd: UVData object
+        uvdata object of data to compute sky-frate limits for.
+    keys: list of antpairpol tuples, optional
+        list of antpairpols to generate sky fringe-rate centers and widths for.
+        Default is None -> use all keys in self.data.
+    frate_standoff: float, optional
+        Additional fringe-rate standoff in mHz to add to Omega_E b_{EW} nu/c for fringe-rate inpainting.
+        default = 0.0.
+    frate_width_multiplier: float, optional
+        fraction of horizon to fringe-rate filter.
+        default is 1.0
+    min_frate_half_width: float, optional
+        minimum fringe-rate to filter, regardless of baseline length in mHz.
+        Default is 0.025
+
+    Returns
+    -------
+    frate_centers: DataContainer object,
+        DataContainer with the center fringe-rate of each baseline in keys in units of mHz.
+    frate_half_widths: DataContainer object
+        DataContainer with the half widths of each fringe-rate window around the frate_centers in units of mHz.
+
+    """
+    if keys is None:
+        keys = uvd.get_antpairpols()
+    antpos, antnums = uvd.get_ENU_antpos()
+    sinlat = np.sin(np.abs(uvd.telescope_location_lat_lon_alt[0]))
+    frate_centers = {}
+    frate_half_widths = {}
+
+    # compute maximum fringe rate dict based on baseline lengths.
+    for k in keys:
```

Comment:
can you leave a comment on the eqn. number of Parsons 2016 you are using here?

Reply:
The expressions used for max / min fringe-rates don't appear in Parsons 2016 (they are derived from expressions in that paper though). I've added a link to this memo:

https://www.overleaf.com/read/chgpxydbhfhk

Reply:
i don't see the link

Reply:
It's here:

 https://github.com/HERA-Team/hera_cal/blob/6a1055b0a6edd46b498300e3272c88844f2cdbbc/hera_cal/frf.py#L65

for some reason it isn't showing up in the diff 🤔

Reply:
now it's showing up?